File: frontend/lib/models/scan.dart
class Scan {
  final int id;
  final String crop;
  final String plague;
  final String severity;
  final double confidence;
  final String? location;
  final int? farmId;
  final DateTime? createdAt;
  final bool shareWithTech;
  final String? techStatus;
  final String? correctedPlague;
  final String? techNotes;

  Scan({
    required this.id,
    required this.crop,
    required this.plague,
    required this.severity,
    this.confidence = 0.0,
    this.location,
    this.farmId,
    this.createdAt,
    this.shareWithTech = false,
    this.techStatus,
    this.correctedPlague,
    this.techNotes,
  });

  factory Scan.fromJson(Map<String, dynamic> json) {
    return Scan(
      id: json["id"] as int,
      crop: json["crop"] as String,
      plague: json["plague"] as String,
      severity: json["severity"] as String,
      confidence: (json["confidence"] as num?)?.toDouble() ?? 0.0,
      location: json["location"] as String?,
      farmId: json["farm_id"] as int?,
      createdAt: json["created_at"] != null ? DateTime.parse(json["created_at"] as String) : null,
      shareWithTech: json["share_with_tech"] as bool? ?? false,
      techStatus: json["tech_status"] as String?,
      correctedPlague: json["corrected_plague"] as String?,
      techNotes: json["tech_notes"] as String?,
    );
  }

  bool get isVerifiedByTech => techStatus == "confirmed" || techStatus == "corrected";

  bool get isRejectedByTech => techStatus == "rejected";

  bool get isUnverified => !isVerifiedByTech && !isRejectedByTech;

  String get effectivePlague => correctedPlague ?? plague;
}

File: backend/app/models/scan.py
from datetime import datetime

from sqlalchemy import Boolean, Column, DateTime, Float, ForeignKey, Integer, String
from sqlalchemy.orm import relationship

from app.db.base import Base


class Scan(Base):
    __tablename__ = "scans"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    crop = Column(String, nullable=False)
    plague = Column(String, nullable=False)
    confidence = Column(Float, default=0.0)
    severity = Column(String, nullable=False)
    location = Column(String, nullable=True)
    farm_id = Column(Integer, ForeignKey("farms.id"), nullable=True, index=True)
    created_at = Column(DateTime(timezone=True), default=datetime.utcnow, nullable=False, index=True)
    image_path = Column(String(255), nullable=True)
    share_with_tech = Column(Boolean, default=False, nullable=False, index=True)
    tech_status = Column(String(20), nullable=True, index=True)
    corrected_plague = Column(String(50), nullable=True)
    tech_notes = Column(String(500), nullable=True)
    validated_by_id = Column(Integer, ForeignKey("users.id"), nullable=True)
    validated_at = Column(DateTime(timezone=True), nullable=True)
    user = relationship("User", foreign_keys=[user_id])
    validated_by = relationship("User", foreign_keys=[validated_by_id])
